File: PSW-projekat/PZL-Angular/src/app/modules/hospital/model/roomDto.ts
import { Bed } from "./bed";
import { BedDto } from "./bedDto";
import { Blood } from "./blood";
import { Medicine } from "./medicine";
import { RoomFloor } from "./roomFloor";
import { RoomName } from "./roomName";
import { RoomType } from "./roomType";

export class RoomDto {
    id: number = 0;
    number: RoomName;
    floor: RoomFloor;
    roomType: RoomType;
    bedDtos: BedDto[];

    public constructor(id: any, number: any, floor: any, roomType: any, bedDtos: any) {
        this.id = id;
        this.number = number;
        this.floor = floor;
        this.roomType = roomType;
        this.bedDtos = bedDtos;

    }

    
}

File: PSW-projekat/hospital-be-main/src/HospitalLibrary/Core/DTOs/RoomDto.cs
using System.Collections.Generic;
using HospitalLibrary.Core.Model;
using HospitalLibrary.Core.Model.Enums;

namespace HospitalLibrary.Core.DTOs
{
    public class RoomDto
    {
        public int Id { get; set; }
        public RoomName Number { get; set; }
        //[Range(1, 10)]

        public RoomFloor Floor { get; set; }
        public RoomType RoomType { get; set; }
        public virtual ICollection<BedDto> BedDtos { get; set; }

        public RoomDto() { }
        public RoomDto(int id, RoomName number, RoomFloor floor, RoomType roomType, ICollection<BedDto> bedDtos)
        {
            Id = id;
            Number = number;
            Floor = floor;
            RoomType = roomType;
            BedDtos = bedDtos;
        }
    }
}
